cli/local/docker_spec.go

```diff
@@ -295,13 +295,28 @@ func deployTensorFlowContainers(api *spec.API, awsClient *aws.Client) error {
 		Mounts:    mounts,
 	}
 
+	envVars := []string{}
+	cmdArgs := []string{
+		"--port=" + _tfServingPortStr,
+		"--model_config_file=" + _tfServingEmptyModelConfig,
+	}
+	if api.Predictor.BatchSize != nil && api.Predictor.BatchTimeout != nil {
+		envVars = append(envVars,
+			"TF_BATCH_SIZE="+s.Int32(*api.Predictor.BatchSize),
+			"TF_BATCH_TIMEOUT_MICROS="+s.Int64(int64(*api.Predictor.BatchTimeout*1000000)),
+			"TF_NUM_BATCHED_THREADS="+s.Int32(api.Predictor.ProcessesPerReplica),
```

Comment:
Yes, it has to be `api.Predictor.ProcessesPerReplica` because this is encouraged to be set to the number of available CPU cores. This is not tied to the replica's concurrency level. More on this [here](https://github.com/tensorflow/serving/tree/master/tensorflow_serving/batching#performance-tuning).

Also, regarding @vishalbollu's concern about having the maximum batch size set to a value higher than `api.Predictor.ProcessesPerReplica * api.Predictor.ThreadsPerProcess`, yes, I confirm that is validated beforehand. If the maximum batch size is greater than `api.Predictor.ProcessesPerReplica * api.Predictor.ThreadsPerProcess`, then an `ErrorInsufficientBatchConcurrencyLevel` error is returned.